.github/workflows/compliance.yml

```diff
@@ -0,0 +1,112 @@
+name: Compliance
+
+on: pull_request
+
+jobs:
+  compliance_job:
+    runs-on: ubuntu-latest
+    name: Run compliance checks on patch series (PR)
+    steps:
+    - name: Checkout the code
+      uses: actions/checkout@v1
+
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-doc-pip
+
+    - name: Install python dependencies
+      run: |
+        pip3 install setuptools
+        pip3 install wheel
+        pip3 install python-magic junitparser gitlint pylint pykwalify
+        pip3 install west
+
+    - name: Run Compliance Tests
+      id: compliance
+      run: |
+        export PATH=$PATH:~/.local/bin
+        export ZEPHYR_BASE=$PWD
+        git config --global user.email "you@example.com"
+        git config --global user.name "Your Name"
+        git rebase origin/master
```

Comment:
What about pull requests to non-master branch? (e.g. backports)?

Should this use `${GITHUB_REF}` or something similar instead?

Reply:
yeah, forgot about this, let me fix it.